internal/node/advancer/service/service.go

```diff
@@ -0,0 +1,64 @@
+// (c) Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+package service
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+	"sync/atomic"
+	"time"
+)
+
+type Service interface {
+	Run(context.Context) error
+}
+
+type Poller struct {
+	name       string
+	service    Service
+	shouldStop atomic.Bool
+	ticker     *time.Ticker
+}
+
+var ErrInvalidPollingInterval = errors.New("polling interval must be greater than zero")
+
+func NewPoller(name string, service Service, pollingInterval time.Duration) (*Poller, error) {
+	if pollingInterval <= 0 {
+		return nil, ErrInvalidPollingInterval
+	}
+	ticker := time.NewTicker(pollingInterval)
+	return &Poller{name: name, service: service, ticker: ticker}, nil
+}
+
+func (poller *Poller) Start(ctx context.Context, ready chan<- struct{}) error {
+	ready <- struct{}{}
+
+	slog.Info(fmt.Sprintf("%s: started", poller.name))
```

Comment:
I'm keeping it because it makes more sense to add this here, given that the Poller will most likely be used by all services. I did change the level to Debug.